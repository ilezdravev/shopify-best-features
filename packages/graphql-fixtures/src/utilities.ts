export function chooseNull() {
  return Math.random() < 0.5;
}

export function randomFromArray<T>(array: T[] | ReadonlyArray<T>) {
  return array[Math.floor(Math.random() * array.length)];
}
