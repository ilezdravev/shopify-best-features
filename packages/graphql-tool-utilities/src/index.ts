export * from './ast';

export {
  getGraphQLProjectForSchemaPath,
  getGraphQLProjectIncludedFilePaths,
  getGraphQLProjects,
  getGraphQLSchemaPaths,
} from './config';
export * from './utilities';
