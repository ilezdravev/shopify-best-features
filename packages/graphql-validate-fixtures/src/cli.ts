/* eslint no-console: off */

import * as yargs from 'yargs';
import * as glob from 'glob';
import chalk from 'chalk';
import {relative, basename} from 'path';

import {evaluateFixtures} from '.';

const argv = yargs
  .usage('Usage: $0 <fixtures> [options]')
  .option('schema-path', {
    required: true,
    normalize: true,
    type: 'string',
    describe:
      'The path to the JSON file containing a schema instrospection query result',
  })
  .option('operation-paths', {
    required: false,
    normalize: true,
    type: 'string',
    describe:
      'The glob pattern for GraphQL queries and fragments to compare against',
  })
  .option('schema-only', {
    type: 'boolean',
    default: false,
    describe:
      'Validate fixtures only against the GraphQL schema (and not any query or mutation documents)',
  })
  .help().argv;

const hasOperationPaths = Boolean(argv.operationPaths);

evaluateFixtures(
  {
    fixturePaths: glob.sync(argv._[0]),
    operationPaths: hasOperationPaths ? glob.sync(argv.operationPaths) : [],
    schemaPath: argv.schemaPath,
  },
  {
    schemaOnly: argv.schemaOnly || !hasOperationPaths,
  },
)
  .then((evaluations) => {
    let passed = 0;
    let failed = 0;
    let lastFailed = false;

    console.log();

    evaluations.forEach((evaluation) => {
      const relativePath = relative(process.cwd(), evaluation.fixturePath);
      const formattedPath = `${chalk.dim(
        relativePath.replace(basename(relativePath), ''),
      )}${chalk.bold(basename(relativePath))}`;

      if (evaluation.scriptError) {
        failed += 1;
        lastFailed = true;
        console.log(`${chalk.inverse.bold.red(' FAIL ')} ${formattedPath}`);
        console.log(evaluation.scriptError.message);
        if (evaluation.scriptError.stack) {
          console.log(
            chalk.dim(
              evaluation.scriptError.stack
                .replace(evaluation.scriptError.message, '')
                .replace(/^Error:\s*\n/, ''),
            ),
          );
        }
        console.log();
      } else if (evaluation.validationErrors.length === 0) {
        passed += 1;
        lastFailed = false;
        console.log(`${chalk.inverse.bold.green(' PASS ')} ${formattedPath}`);
      } else {
        failed += 1;
        lastFailed = true;

        console.log(`${chalk.inverse.bold.red(' FAIL ')} ${formattedPath}`);
        evaluation.validationErrors.forEach((error) => {
          console.log(`${error.keyPath} ${chalk.red(error.message)}`);
        });
        console.log();
      }
    });

    if (passed + failed === 0) {
      console.log(chalk.bold.yellow('No tests run'));
      return;
    }

    if (!lastFailed) {
      console.log();
    }

    if (passed > 0) {
      console.log(chalk.bold.green(`${passed} passed`));
    }

    if (failed > 0) {
      console.log(chalk.bold.red(`${failed} failed`));
      process.exit(1);
    }
  })
  .catch((error) => {
    console.log();
    console.log(`${chalk.inverse.bold.red(' ERROR ')} ${error.message}`);
    console.log(
      chalk.dim(
        error.stack.replace(error.message, '').replace(/^Error:\s*\n/, ''),
      ),
    );
    process.exit(1);
  });
