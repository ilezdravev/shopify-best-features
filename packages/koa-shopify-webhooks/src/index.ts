export {receiveWebhook} from './receive';
export type {WebhookState} from './receive';

export {registerWebhook} from './register';
export type {Options} from './register';

export {WebhookHeader} from './types';
export type {Topic} from './types';
