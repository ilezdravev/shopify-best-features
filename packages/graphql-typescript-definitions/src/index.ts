import {writeFileSync} from 'fs-extra';
import {Operation, Fragment, AST} from 'graphql-tool-utilities/ast';

import buildAST, {Options} from './ast';
import {printFile} from './print';

export default function graphQLToTypeScriptDefinitions(options: Options) {
  const ast = buildAST(options);
  const fileMap = groupQueriesAndFragmentsByFile(ast);

  Object.keys(fileMap).forEach((path) => {
    const file = fileMap[path];

    const content = printFile(file, ast);
    if (!content) { return; }

    const newFile = `${file.path}.d.ts`;
    writeFileSync(newFile, content);
  });
}

interface File {
  path: string,
  operation?: Operation,
  fragment?: Fragment,
}

interface FileMap {
  [key: string]: File,
}

function groupQueriesAndFragmentsByFile({operations, fragments}: AST): FileMap {
  const map: FileMap = {};

  Object
    .keys(operations)
    .forEach((name) => {
      const operation = operations[name];
      map[operation.filePath] = {
        path: operation.filePath,
        operation,
      };
    });
  
  Object
    .keys(fragments)
    .forEach((name) => {
      const fragment = fragments[name];
      map[fragment.filePath] = {
        path: fragment.filePath,
        fragment,
      };
    });
  
  return map;
}
