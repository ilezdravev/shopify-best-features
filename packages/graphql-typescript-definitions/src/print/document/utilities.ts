import {ucFirst} from 'change-case';
import {GraphQLCompositeType} from 'graphql';
import {Field} from 'graphql-tool-utilities/ast';

export class ObjectStack {
  private seenFields = new Set<string>();

  get name() {
    return this.parentFields
      .map(({responseName}) => ucFirst(responseName))
      .join('');
  }

  constructor(
    _type: GraphQLCompositeType,
    private parentFields: Field[] = [],
  ) {}

  nested(field: Field, type: GraphQLCompositeType) {
    return new ObjectStack(type, [...this.parentFields, field]);
  }

  fragment(type: GraphQLCompositeType) {
    return new ObjectStack(type, this.parentFields);
  }

  sawField(field: Field) {
    this.seenFields.add(field.responseName);
  }

  hasSeenField(field: Field) {
    return this.seenFields.has(field.responseName);
  }
}
