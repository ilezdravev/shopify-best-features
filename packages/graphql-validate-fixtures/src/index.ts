import {readFile, readJSON} from 'fs-extra';
import {isAbsolute, resolve} from 'path';
import {GraphQLSchema, buildClientSchema, Source, parse, concatAST} from 'graphql';
import {compile} from 'graphql-tool-utilities/ast';

import {
  validateFixtureAgainstAST,
  validateFixtureAgainstSchema,
  Validation,
  Fixture,
} from './validate';

export interface Paths {
  fixturePaths: string[],
  operationPaths?: string[],
  schemaPath: string,
}

export interface Options {
  schemaOnly?: boolean,
}

export interface Evaluation extends Validation {
  fixturePath: string,
  scriptError?: Error,
}

export async function evaluateFixtures({fixturePaths, operationPaths = [], schemaPath}: Paths, {schemaOnly = false}: Options = {}): Promise<Evaluation[]>  {
  let schema: GraphQLSchema;

  try {
    const schemaJSON = await readJSON(schemaPath, {encoding: 'utf8'});
    schema = buildClientSchema(schemaJSON.data);
  } catch (error) {
    throw new Error(`Error parsing '${schemaPath}':\n\n${error.message.replace(/Syntax Error GraphQL \(.*?\) /, '')}`);
  }

  if (schemaOnly) {
    return await runForEachFixture(fixturePaths, (fixture) => validateFixtureAgainstSchema(fixture, schema));
  }
  
  const sources = await Promise.all(
    operationPaths.map(async (operationPath) => new Source(await readFile(operationPath, 'utf8'), operationPath))
  );
  const document = concatAST(sources.map((source) => {
    try {
      return parse(source);
    } catch (error) {
      throw new Error(`Error parsing '${source.name}':\n\n${error.message.replace(/Syntax Error.*?\(.*?\) /, '')}`);
    }
  }));
  const ast = compile(schema, document);

  return await runForEachFixture(fixturePaths, (fixture) => validateFixtureAgainstAST(fixture, ast))
}

function runForEachFixture<T extends Partial<Evaluation>>(fixturePaths: string[], runner: (fixture: Fixture) => T): Promise<Evaluation[]> {
  return Promise.all(
    fixturePaths.map(async (fixturePath) => {
      const finalPath = isAbsolute(fixturePath) ? fixturePath : resolve(fixturePath);
      
      try {
        const fixture = await readJSON(finalPath);
        return {
          fixturePath: finalPath,
          ...(runner({path: finalPath, content: fixture}) as any),
        };
      } catch (error) {
        return {
          fixturePath: finalPath,
          scriptError: error,
          validationErrors: [],
        }
      }
    })
  );
}
