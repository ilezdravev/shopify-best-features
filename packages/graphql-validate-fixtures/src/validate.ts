import {dirname, basename} from 'path';
import {
  GraphQLType,
  GraphQLScalarType,
  GraphQLEnumType,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLList,
  GraphQLString,
  GraphQLInt,
  GraphQLFloat,
  GraphQLBoolean,
} from 'graphql';
import {AST, Field} from 'graphql-tool-utilities/ast';

export type KeyPath = string;

export interface Fixture {
  path: string,
  content: any,
}

export interface Error {
  keyPath?: KeyPath,
  message: string,
}

export interface Validation {
  operationName: string,
  operationType: 'query' | 'mutation',
  operationPath?: string,
  errors: Error[],
}

const OPERATION_MARKER = '@operation';

export default function validateFixtureAgainstAST(fixture: Fixture, ast: AST): Validation {
  const fixtureDirectoryName = basename(dirname(fixture.path));
  const operationName = fixture.content[OPERATION_MARKER] || fixtureDirectoryName;
  const operation = ast.operations[operationName];

  if (operation == null) {
    let lookedFor = `'${fixtureDirectoryName}' based on the fixtureâ€™s directory name`
    
    if (fixture.content[OPERATION_MARKER]) {
      lookedFor += `, and '${fixture.content[OPERATION_MARKER]}' based on the '${OPERATION_MARKER}' key`;
    }

    throw new Error([
      `Could not find a matching operation (looked for ${lookedFor}).`,
      `Make sure to put your fixture in a folder named the same as the operation, or add an '${OPERATION_MARKER}' key indicating the operation.`,
      `Available operations: ${Object.keys(ast.operations).join(', ')}`
    ].join(' '));
  }

  const {fields, filePath, operationType} = operation;
  const value = {...fixture.content};
  delete value[OPERATION_MARKER];

  return {
    operationName,
    operationType,
    operationPath: filePath,
    errors: fields.reduce((allErrors: Error[], field) => {
      return allErrors.concat(
        validateValueAgainstFieldDescription(value[field.responseName], field, '', ast)
      );
    }, []),
  };
}

function validateValueAgainstFieldDescription(value: any, fieldDescription: Field, parentKeyPath: string, ast: AST): Error[] {
  const {type, responseName} = fieldDescription;
  const keyPath = updateKeyPath(parentKeyPath, responseName);
  const typeErrors = validateValueAgainstType(value, type, keyPath);

  if (typeErrors.length > 0) {
    return typeErrors;
  }
  
  return Array.isArray(value)
    ? validateListAgainstFieldDescription(value, fieldDescription, fieldDescription.type, keyPath, ast)
    : validateValueAgainstObjectFieldDescription(value, fieldDescription, keyPath, ast);
}

function validateValueAgainstObjectFieldDescription(value: any, fieldDescription: Field, keyPath: string, ast: AST) {
  if (typeof value !== 'object' || Array.isArray(value)) {
    return [];
  }

  const {fields = [], fragmentSpreads = [], type} = fieldDescription;

  let fragmentFields: Field[] = [];

  if (fragmentSpreads) {
    fragmentSpreads
      .map((spread) => ast.fragments[spread])
      .forEach((fragment) => {
        fragment.fields.forEach((field) => {
          if (fields.some(({fieldName}) => fieldName === field.fieldName)) {
            return;
          }

          const isGuaranteedTypeMatch = fragment.possibleTypes.includes(makeTypeNullable(type));

          fragmentFields.push(
            isGuaranteedTypeMatch
              ? field
              : {...field, type: makeTypeNullable(field.type)}
          );
        });
      });
  }

  return validateValueAgainstFields(value, fields.concat(fragmentFields), keyPath, ast);
}

function makeTypeNullable(type: GraphQLType) {
  return type instanceof GraphQLNonNull
    ? type.ofType
    : type;
}

function validateListAgainstFieldDescription(value: any[], fieldDescription: Field, type: GraphQLType, keyPath: string, ast: AST): Error[] {
  const itemType = type instanceof GraphQLNonNull
      ? (type as GraphQLNonNull<GraphQLList<GraphQLType>>).ofType.ofType
      : (type as GraphQLList<GraphQLType>).ofType;

  return value.reduce((allErrors, item, index) => {
    const itemKeyPath = updateKeyPath(keyPath, index);
    const itemTypeErrors = validateValueAgainstType(item, itemType, itemKeyPath);

    if (itemTypeErrors.length > 0) {
      return allErrors.concat(itemTypeErrors);
    }

    return Array.isArray(item)
      ? allErrors.concat(validateListAgainstFieldDescription(item, fieldDescription, itemType, itemKeyPath, ast))
      : allErrors.concat(validateValueAgainstObjectFieldDescription(item, fieldDescription, itemKeyPath, ast));
  }, []);
}

function validateValueAgainstFields(value: {[key: string]: any}, fields: Field[], keyPath: KeyPath, ast: AST) {
  const finalValue = value || {};

  const excessFields = Object
    .keys(finalValue)
    .filter((key) => fields.every(({responseName}) => key !== responseName))
    .map((key) => error(keyPath, `has key '${key}' not present in the query`));

  return fields.reduce((allErrors, field) => {
    return allErrors.concat(
      validateValueAgainstFieldDescription(finalValue[field.responseName], field, keyPath, ast)
    );
  }, excessFields);
}

function validateValueAgainstType(value: any, type: GraphQLType, keyPath: KeyPath): Error[] {
  if (type instanceof GraphQLNonNull) {
    return value == null
      ? [error(keyPath, `should be non-null but was ${String(value)}`)]
      : validateValueAgainstType(value, type.ofType, keyPath);
  }

  if (value == null) { return []; }

  const valueType = typeof value;

  if (type instanceof GraphQLObjectType) {
    return valueType === 'object'
      ? []
      : [error(keyPath, `should be an object but was a ${valueType}`)];
  }

  if (type instanceof GraphQLList) {
    if (!Array.isArray(value)) {
      return [error(keyPath, `should be an array, but was a ${valueType}`)];
    }

    return Array.isArray(value)
      ? []
      : [error(keyPath, `should be an array, but was a ${valueType}`)];
  }

  if (type === GraphQLString) {
    return valueType === 'string'
      ? []
      : [error(keyPath, `should be a string but was a ${valueType}`)];
  } else if (type === GraphQLInt) {
    if (typeof value === 'number') {
      return Number.isInteger(value)
        ? []
        : [error(keyPath, 'should be an integer but was a float')]
    }

    return [error(keyPath, `should be an integer but was a ${valueType}`)];
  } else if (type === GraphQLFloat) {
    if (typeof value === 'number') {
      return Number.isInteger(value)
        ? [error(keyPath, 'should be a float but was an integer')]
        : [];
    }

    return [error(keyPath, `should be a float but was a ${valueType}`)];
  } else if (type === GraphQLBoolean) {
    return typeof value === 'boolean'
      ? []
      : [error(keyPath, `should be a boolean but was a ${valueType}`)];
  }

  if (type instanceof GraphQLScalarType) {
    return type.parseValue(value) != null
      ? []
      : [error(keyPath, `value does not match scalar ${type.name}`)];
  }

  if (type instanceof GraphQLEnumType) {
    return type.parseValue(value) != null
      ? []
      : [error(keyPath, `value does not match enum ${type.name} (available values: ${type.getValues().map((enumValue) => enumValue.value).join(', ')})`)];
  }

  return [];
}

function updateKeyPath(keyPath: KeyPath, newKey: string | number) {
  if (typeof newKey === 'number') {
    return `${keyPath}[${newKey}]`;
  }

  return keyPath ? `${keyPath}.${newKey}` : newKey;
}

function error(keyPath: KeyPath, message: string): Error {
  return {keyPath, message};
}
