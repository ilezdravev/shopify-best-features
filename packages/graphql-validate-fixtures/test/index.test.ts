import * as glob from 'glob';
import {join} from 'path';
import {stripFullFilePaths} from '../../../test/utilities';
import {evaluateFixtures, Options} from '../src';

const rootFixtureDirectory = join(__dirname, 'fixtures');

describe('evaluateFixtures()', () => {
  it('handles fixtures without errors', async () => {
    expect(await evaluateFixturesForFixturePath('all-clear')).toMatchSnapshot();
  });

  it('handles fixtures with errors', async () => {
    expect(
      await evaluateFixturesForFixturePath('fixture-errors'),
    ).toMatchSnapshot();
  });

  it('handles fixtures that are invalid json', async () => {
    expect(
      await evaluateFixturesForFixturePath('fixture-invalid'),
    ).toMatchSnapshot();
  });

  it('throws an error when the schema is not found', async () => {
    await expect(
      evaluateFixturesForFixturePath('missing-schema'),
    ).rejects.toMatchSnapshot();
  });

  it('throws an error when there are malformed GraphQL documents', async () => {
    await expect(
      evaluateFixturesForFixturePath('malformed-query'),
    ).rejects.toMatchSnapshot();
  });

  describe('schemaOnly', () => {
    it('validates against the schema only', async () => {
      expect(
        await evaluateFixturesForFixturePath('fixture-errors', {
          schemaOnly: true,
        }),
      ).toMatchSnapshot();
    });

    it('does not fail on malformed GraphQL documents', async () => {
      expect(
        await evaluateFixturesForFixturePath('malformed-query', {
          schemaOnly: true,
        }),
      ).toMatchSnapshot();
    });
  });
});

async function evaluateFixturesForFixturePath(
  fixture: string,
  options?: Options,
) {
  try {
    const result = await evaluateFixtures(detailsForFixture(fixture), options);
    return stripFullFilePaths(result);
  } catch (error) {
    throw stripFullFilePaths(error);
  }
}

function detailsForFixture(fixture: string) {
  const fixtureDirectory = join(rootFixtureDirectory, fixture);
  const schemaPath = join(fixtureDirectory, 'schema.json');
  const fixturePaths = glob.sync(join(fixtureDirectory, 'fixtures/**/*.json'));
  const operationPaths = glob.sync(
    join(fixtureDirectory, 'queries/**/*.graphql'),
  );

  return {
    schemaPath,
    fixturePaths,
    operationPaths,
  };
}
