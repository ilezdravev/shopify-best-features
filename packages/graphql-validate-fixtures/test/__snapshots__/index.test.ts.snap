// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`evaluateFixtures() handles fixtures that are invalid json 1`] = `
Array [
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/fixtures/another-fixture.json",
    "scriptError": [SyntaxError: packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/fixtures/another-fixture.json: Unexpected string in JSON at position 35],
    "validationErrors": Array [],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/fixtures/MyQuery/fixture.json",
    "operationName": "MyQuery",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/fixture-invalid/queries/Query.graphql",
    "operationType": "query",
    "validationErrors": Array [],
  },
]
`;

exports[`evaluateFixtures() handles fixtures with errors 1`] = `
Array [
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/fixtures/another-fixture.json",
    "operationName": "AnotherQuery",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/queries/Another.graphql",
    "operationType": "query",
    "validationErrors": Array [
      Object {
        "keyPath": "name",
        "message": "should be a string but was a number",
      },
      Object {
        "keyPath": "age",
        "message": "should be non-null but was undefined",
      },
    ],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/fixtures/MyQuery/fixture.json",
    "operationName": "MyQuery",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/queries/Query.graphql",
    "operationType": "query",
    "validationErrors": Array [],
  },
]
`;

exports[`evaluateFixtures() handles fixtures without errors 1`] = `
Array [
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/fixtures/another-another-fixture.json",
    "operationName": "Another",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/queries/Another.graphql",
    "operationType": "query",
    "validationErrors": Array [],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/fixtures/another-fixture.json",
    "operationName": "AnotherQuery",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/queries/Another.graphql",
    "operationType": "query",
    "validationErrors": Array [],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/fixtures/MyQuery/fixture.json",
    "operationName": "MyQuery",
    "operationPath": "packages/graphql-validate-fixtures/test/fixtures/all-clear/queries/Query.graphql",
    "operationType": "query",
    "validationErrors": Array [],
  },
]
`;

exports[`evaluateFixtures() schemaOnly does not fail on malformed GraphQL documents 1`] = `
Array [
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/malformed-query/fixtures/another-fixture.json",
    "validationErrors": Array [],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/malformed-query/fixtures/MyQuery/fixture.json",
    "validationErrors": Array [],
  },
]
`;

exports[`evaluateFixtures() schemaOnly validates against the schema only 1`] = `
Array [
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/fixtures/another-fixture.json",
    "validationErrors": Array [
      Object {
        "keyPath": "name",
        "message": "should be a string but was a number",
      },
    ],
  },
  Object {
    "fixturePath": "packages/graphql-validate-fixtures/test/fixtures/fixture-errors/fixtures/MyQuery/fixture.json",
    "validationErrors": Array [],
  },
]
`;

exports[`evaluateFixtures() throws an error when the schema is not found 1`] = `
[Error: Error parsing 'packages/graphql-validate-fixtures/test/fixtures/missing-schema/schema.json':

ENOENT: no such file or directory, open 'packages/graphql-validate-fixtures/test/fixtures/missing-schema/schema.json']
`;

exports[`evaluateFixtures() throws an error when there are malformed GraphQL documents 1`] = `
[Error: Error parsing 'packages/graphql-validate-fixtures/test/fixtures/malformed-query/queries/Another.graphql':

Syntax Error: Expected {, found Name "name"]
`;
